test/e2e/kcp_upgrade.go

```diff
@@ -92,20 +92,34 @@ func KCPUpgradeSpec(ctx context.Context, inputGetter func() KCPUpgradeSpecInput)
 			WaitForMachineDeployments:    input.E2EConfig.GetIntervals(specName, "wait-worker-nodes"),
 		})
 
-		By("Upgrading Kubernetes, DNS, kube-proxy, and etcd versions")
+		By("Upgrading Kubernetes & kube-proxy to 1.17, CoreDNS to 1.6.5, and etcd to 3.4.3")
 		framework.UpgradeControlPlaneAndWaitForUpgrade(ctx, framework.UpgradeControlPlaneAndWaitForUpgradeInput{
 			ClusterProxy: input.BootstrapClusterProxy,
 			Cluster:      cluster,
 			ControlPlane: controlPlane,
-			//Valid image tags for v1.17.2
+			// Valid image tags for v1.17.2
 			EtcdImageTag:                "3.4.3-0",
-			DNSImageTag:                 "1.6.6",
+			DNSImageTag:                 "1.6.5",
 			KubernetesUpgradeVersion:    "v1.17.2",
 			WaitForMachinesToBeUpgraded: input.E2EConfig.GetIntervals(specName, "wait-machine-upgrade"),
 			WaitForDNSUpgrade:           input.E2EConfig.GetIntervals(specName, "wait-machine-upgrade"),
 			WaitForEtcdUpgrade:          input.E2EConfig.GetIntervals(specName, "wait-machine-upgrade"),
 		})
 
+		By("Upgrading Kubernetes & kube-proxy to 1.18, CoreDNS to 1.6.7, and etcd to 3.4.3")
```

Comment:
Is it required to do the upgrade twice?
an upgrade is a time expensive operation and this will impact all the PRs...

Reply:
I would have thought we would want to exercise etcd and coredns upgrades, but there wasn't a change from 1.17 to 1.18 for etcd.

Reply:
Since upgrade takes too long, one of the upgrades we can do ControlPlaneMachineCount=1.
It is a test scenario we want to add anyways.

Reply:
I'm not opposed. @vincepri any thoughts.
I think I might want do any of these optimisations in another PR.